src/accelerate/big_modeling.py

```diff
@@ -379,6 +384,25 @@ def dispatch_model(
         )
         # Attaching the hook may break tied weights, so we retie them
         retie_parameters(model, tied_params)
+
+        # add warning to cuda and to method
+        def add_warning(fn):
+            @wraps(fn)
+            def wrapper(*args, **kwargs):
+                logger.warning(
+                    "You can't use the model anymore for training or inference as you moved the model."
+                    "You should not move the model when it is dispatched on multiples devices. "
```

Comment:
This warning is not super clear. I would say something more like "You can't move a model when it is dispatched on multiple device." The model can still be used for inference and training.

Reply:
changed !